signature-specification.md

```diff
@@ -0,0 +1,215 @@
+* **`crit`**(*array of strings*): This REQUIRED property lists the headers that implementation MUST understand and process. The value MUST be `["cty"]`.
+
+**UnprotectedHeaders**:
+Notary v2 supports only two unprotected headers: timestamp and x5c.
+```
+{
+   "timestamp": "<Base64Url(TimeStampToken)>",
+   "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
+}
+```
+* **`timestamp`**(*string*): This OPTIONAL property is used to store time stamp token. Only [RFC3161]([rfc3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2)) compliant TimeStampToken are supported.
+* **`x5c`**(*array of strings*): This REQUIRED property contains the list of X.509 certificate or certificate chain([RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)) corresponding to the key used to digitally sign the JWS.
+
+**Signature**:
+In JWS signature is calculated by combining JWSPayload and protected headers. The process is described below:
+1. Compute the Base64Url value of ProtectedHeaders.
+2. Compute the Base64Url value of JWSPayload.
+3. Build message to be signed by concatenating the values generated in step 1 and step 2 using '.'
+`ASCII(BASE64URL(UTF8(ProtectedHeaders)) ‘.’ BASE64URL(JWSPayload))`
+4. Compute the signature on the message constructed in the previous step by using the signature algorithm defined in the corresponding header element: alg.
+5. Compute the Base64Url value of the signature produced in the previous step. This is the value of the signature property used in the signature envelope.
+
+**Signature Envelope**:
+The final signature envelope comprises of Claims, ProtectedHeaders, UnprotectedHeaders, and signature.
+
+Since Notary v2 restricts one signature per signature envelope, the compliant signature envelope MUST be in flattened JWS JSON format.
+```json
+{
+   "payload": "<Base64Url(JWSPayload)>",
+   "protected": "<Base64Url(ProtectedHeaders)>",
+   "header": {
+     "timestamp": "<Base64Url(TimeStampToken)>",
+     "x5c": ["<Base64(DER(leafCert))>", "<Base64(DER(intermediateCACert))>", "<Base64(DER(rootCert))>"]
+   },
+   "signature": "Base64Url( sign( ASCII( <Base64Url(ProtectedHeader)>.<Base64Url(JWSPayload)> )))"  
+}
+```
+
+**Implementation Constraints**
+Notary v2 implementation MUST enforce the following constraints on signature generation and verification:
+1. `alg` header value MUST NOT be `none` or any symmetric-key algorithm such as `HMAC`.
+2. `alg` header value MUST be same as that of signature algorithm identified using signing certificate's public key algorithm and size.
+3. `alg` header values for various signature algorithms:
+   | Signature Algorithm    | "alg" Param Value |
+   | --------    | -------- |
+   | ECDSA on secp256r1 with SHA-256 | ES256     |
```

Comment:
Can we have a more friendly name for `secp256r1` from SECG? like `P-256` from NIST?

Also, we need to provide a reference for readers. [RFC 8422 Appendix A](https://datatracker.ietf.org/doc/html/rfc8422#appendix-A) is a good one.

Reply:
Not all algos and curves are in NIST and that's the reason refrained from using NIST curves. 
Added reference to RFC 8422 Appendix A  for discovery.